File: dotnet/src/Pactman.NonprofitCheckPlus/Models/AroeSource.cs
using System.Collections.Generic;
using System.Text.Json;

namespace Pactman.NonprofitCheckPlus.Models
{
    /// <summary>IRS Automatic Revocation of Exemption findings.</summary>
    public sealed class AroeSource : SourceView
    {
        /// <summary>Initializes the view over the projected fields.</summary>
        /// <param name="fields">The fields copied from the organization.</param>
        public AroeSource(IEnumerable<KeyValuePair<string, JsonElement>> fields)
            : base(fields)
        {
        }

        /// <summary>The IRS automatic revocation code, when the exemption was revoked.</summary>
        public string? RevocationCode => GetString("revocation_code");

        /// <summary>The date the exemption was automatically revoked.</summary>
        public string? RevocationDate => GetString("revocation_date");

        /// <summary>The date the exemption was reinstated, when it was.</summary>
        public string? ReinstatementDate => GetString("reinstatement_date");
    }
}

File: java/sdk/src/main/java/org/pactman/nonprofitcheckplus/models/AroeSource.java
package org.pactman.nonprofitcheckplus.models;

import java.util.Map;

/** IRS Automatic Revocation of Exemption findings. */
public final class AroeSource extends SourceView {

    /**
     * Initializes the view over the projected fields.
     *
     * @param fields the fields copied from the organization.
     */
    public AroeSource(Map<String, Object> fields) {
        super(fields);
    }

    /**
     * The IRS automatic revocation code, when the exemption was revoked.
     *
     * @return the code, or {@code null} when the API returned none.
     */
    public String getRevocationCode() {
        return getString("revocation_code");
    }

    /**
     * The date the exemption was automatically revoked.
     *
     * @return the date as the API formats it, or {@code null}.
     */
    public String getRevocationDate() {
        return getString("revocation_date");
    }

    /**
     * The date the exemption was reinstated, when it was.
     *
     * @return the date as the API formats it, or {@code null}.
     */
    public String getReinstatementDate() {
        return getString("reinstatement_date");
    }
}
